flex/engines/graph_db/database/wal.cc

```diff
@@ -70,28 +198,82 @@ void WalWriter::append(const char* data, size_t length) {
   file_used_ += length;
 
   if (static_cast<size_t>(write(fd_, data, length)) != length) {
-    LOG(FATAL) << "Failed to write wal file " << strerror(errno);
+    LOG(ERROR) << "Failed to write wal file " << strerror(errno);
```

Comment:
The `append()` is refactored into returning a bool value, and check outside the function.